### DISCUSSION TITLE
Solara without any internet access

### DISCUSSION THREAD

**Discussion by valoudi** [widgetti/solara#116] [Q&A]
Hi guys ! 
I discover Solara at Jupyter Con (with a great Talk !) 

This morning, I try to execute a small exemple of solara application. 

I've : pip install solara / create an app.py (from your doc) / run : solara run app.py

--> I've got this error when i go onto the website  : 

`requests.exceptions.ConnectionError: HTTPSConnectionPool(host='cdn.jsdelivr.net', port=443): Max retries exceeded with url: /npm/@widgetti/solara-vuetify-app@4.0.0/dist/main.css (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8d6d9c7a30>: Failed to establish a new connection: [Errno -2] Name or service not known'))`

--> On my server, I can't access to internet, to use Pip, i use Nexus to redirect pip install to a local repository of Pypi 

How can I use Solara without Internet, and without the .css that solara try to download ? 

Thanks for your help,

Regards,

Valentin

**Comment by maartenbreddels** [comment]
Hi there!

Thank you very much! Glad you liked my talk.

Short answer, `pip install "solara[assets]"`

But I took the opportunity to document this in this commit: https://github.com/widgetti/solara/commit/9496712ec5510ad74564bb1808c1b96e6cb7fd10
(it should be on the website soon).
Any feedback on that commit is appreciated. Let me know if this helps you and if the instructions are clear and accurate.

Regards,

Maarten

**Comment by valoudi** [comment]
Hi :) 

I think it solves a first error ... 

Now It tries to acces to : 

` requests.exceptions.ConnectionError: HTTPSConnectionPool(host='cdn.jsdelivr.net', port=443): Max retries exceeded with url: /npm/mermaid@9.1.7/dist/mermaid.js (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fb0674510d0>: Failed to establish a new connection: [Errno -2] Name or service not known')) `

I've uploaded a full stacktrace :) 
[error.txt](https://github.com/widgetti/solara/files/11565844/error.txt)

Thanks for your help :) 

Regards,

Valentin

**Reply by maartenbreddels** [reply]
See https://github.com/widgetti/solara/issues/75 (it shouldn't be an issue, solara will work, but this does need fixing)

**Reply by valoudi** [reply]
Huum, It's an issue :) 

With pip install "solara[assets]" the logo of Solara with Loading app is now here but if I check on Google Chrome Console : 

![errorCDN](https://github.com/widgetti/solara/assets/32192595/e06edcaf-1555-4827-9fde-0c9d809a8eca)


And It's loading loading loading ... without success :) 

Thanks again, 

Valentin

**Reply by maartenbreddels** [reply]
do you see anything in the python output?

**Reply by maartenbreddels** [reply]
Send me an email at maartenbreddels@widgetti.io if you want to do a video call with me to help you debug this.

**Reply by Sherlaut** [reply]
Hi, I have the same issue. Did you manage to find a solution ?

Thanks,
Charlotte

**Reply by maartenbreddels** [reply]
Hi Charlotte,

What exactly is your issue? Do you have an error message for us?

Regards,

Maarten

**Reply by Sherlaut** [reply]
I need to use solara without internet, so I try to pip install "solara[assets]", it didn't change the issue. 
This is the error I have : 

File "P:\desktop_charlotte\Python311\Python311\Lib\site-packages\requests\adapters.py", line 513, in send
    raise ProxyError(e, request=request)
requests.exceptions.ProxyError: HTTPSConnectionPool(host='cdn.jsdelivr.net', port=443): Max retries exceeded with url: /npm/requirejs@2.3.6/require.js (Caused by ProxyError('Unable to connect to proxy', NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000002504A9F5A10>: Failed to establish a new connection

**Reply by maartenbreddels** [reply]
This might be a windows specific issue. I tried to reproduce this by setting the `HTTP_PROXY` and `HTTPS_PROXY` environment variables to something non-existing.
If I then not install solara-assets, I indeed get an error like this. If I install the assets I do not see this. 
This could be a windows specific issue. Would you mind opening a new issue for this (not a discussion), so we do not lose track of this?
